crates/bevy_ecs/src/storage/blob_vec.rs

```diff
@@ -83,33 +83,32 @@ impl BlobVec {
 
     pub fn reserve_exact(&mut self, additional: usize) {
         let available_space = self.capacity - self.len;
-        if available_space < additional {
-            self.grow_exact(additional - available_space);
+        if available_space < additional && self.item_layout.size() > 0 {
```

Comment:
If the item size is 0, the capacity is set to `usize::MAX` and thus `available_space` would always be at least as large as `additional` and thus `grow_exact` would never be called.

Reply:
Maybe `grow_exact` should take the desired capacity? Then we would do `let new_capacity = self.len + additional;` and `self.capacity < new_capacity` would can't ever return true for ZST's even if `self.len` is non-zero as `self.len + additional` would overflow when reserving `usize::MAX` elements.

Reply:
I think that API would be significantly easier to use correctly as well.